homeassistant/components/template/trigger_entity.py

```diff
@@ -0,0 +1,161 @@
+"""Trigger entity."""
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+from homeassistant.const import (
+    CONF_DEVICE_CLASS,
+    CONF_ENTITY_PICTURE_TEMPLATE,
+    CONF_FRIENDLY_NAME,
+    CONF_FRIENDLY_NAME_TEMPLATE,
+    CONF_ICON_TEMPLATE,
+    CONF_UNIQUE_ID,
+    CONF_UNIT_OF_MEASUREMENT,
+    CONF_VALUE_TEMPLATE,
+)
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers import template, update_coordinator
+from homeassistant.helpers.entity import async_generate_entity_id
+
+from . import TriggerUpdateCoordinator
+from .const import CONF_ATTRIBUTE_TEMPLATES, CONF_AVAILABILITY_TEMPLATE
+
+
+class TriggerEntity(update_coordinator.CoordinatorEntity):
+    """Sensor entity based on trigger data."""
+
+    domain = ""
+    extra_template_keys: tuple | None = None
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        coordinator: TriggerUpdateCoordinator,
+        device_id: str,
+        config: dict,
+    ):
+        """Initialize the entity."""
+        super().__init__(coordinator)
+
+        self.entity_id = async_generate_entity_id(
+            self.domain + ".{}", device_id, hass=hass
+        )
+
+        self._name = config.get(CONF_FRIENDLY_NAME, device_id)
+
+        entity_unique_id = config.get(CONF_UNIQUE_ID)
+
+        if entity_unique_id is None and coordinator.unique_id:
+            entity_unique_id = device_id
+
+        if entity_unique_id and coordinator.unique_id:
+            self._unique_id = f"{coordinator.unique_id}-{entity_unique_id}"
+        else:
+            self._unique_id = entity_unique_id
+
+        self._config = config
+
+        self._to_render = [
+            itm
+            for itm in (
+                CONF_VALUE_TEMPLATE,
+                CONF_ICON_TEMPLATE,
+                CONF_ENTITY_PICTURE_TEMPLATE,
+                CONF_FRIENDLY_NAME_TEMPLATE,
+                CONF_AVAILABILITY_TEMPLATE,
+            )
+            if itm in config
+        ]
+
+        if self.extra_template_keys is not None:
+            self._to_render.extend(self.extra_template_keys)
+
+        self._rendered = {}
+
+    @property
+    def name(self):
+        """Name of the entity."""
+        if (
+            self._rendered is not None
+            and (name := self._rendered.get(CONF_FRIENDLY_NAME_TEMPLATE))
+            and name is not None
```

Comment:
This line doesn't seem needed.
```suggestion
```

Reply:
When the entity is added to Home Assistant, nothing is rendered yet. This line will make sure we fallback to have at least the configured `object_id`. as name.

Reply:
Doesn't the line above with the walrus operator already catch that?

Reply:
You are right.